### ISSUE TITLE
Code Highlighting Example has jumpy scroll behaviour

### ISSUE THREAD

**Issue by cobbweb** [ianstormtaylor/slate#279]
Scrolling goes all over the place once the editor height exceeds the viewport height.

Steps to reproduce:
1. Click into the Code Block (anywhere will do)
2. Hold Enter to insert _heaps_ of blank lines
3. After enough line breaks are added, the scrolling position jumps with every added/removed line break

<div align="center"><img src="http://g.recordit.co/CqngPARGQe.gif"></div>

Mainly raising this because I'm implementing my own code highlighting block and I've got the same issue.

**Comment by ianstormtaylor** [comment]
For anyone else jumping into this:

- The code that does the scrolling is [here](https://github.com/ianstormtaylor/slate/blob/master/src/utils/scroll-to.js). It's copied from `element-scroll-to` but it accounts for `window` being inside an iframe.
- The logic that triggers the scrolling is [here](https://github.com/ianstormtaylor/slate/blob/master/src/components/node.js#L193).

My guess is that it's a simple bug in `scroll-to` not handling if the cursor is already in view. Would gladly accept a PR that fixes this!

That said, the "correct" way to do this I realized would actually not have anything to do with the node's DOM element, and instead it would scroll such that the cursor itself is centered. Would super gladly accept a PR that switches us to handle it the "correct" way.

**Comment by gusano** [comment]
I have the same issue (rich text example): if one text paragraph is bigger than the screen and I hit BACKSPACE at the end of it, it jumps to the top.

**Comment by pa-nam** [comment]
@everyone @ianstormtaylor Hello.
Seem like now Im having an issue which relate to this.
I have a very tall inline component, when ever I insert a newline at right before or after that inline component, it will scroll to the top where that tall inline is.
I tried some debugging in js and found out that the native selection in the function `updateScroll` in `/component/node.js` does not have a correct value when `updateScroll` is being called.
I guessed the native selection will be changed after finish the frame for componentDidMount or componentDidUpdate, then I went to node_modules and tried myself to add setTimeout to wrap the scrolling logic and the scrolling worked as expected. I created a pull here which includes the change I made: #847. 
Waiting to be accepted (or fixed).
Thanks guys 👍
